tests/vfs_open_mode_tests.cpp

```diff
@@ -0,0 +1,99 @@
+#include <sqlite_orm/sqlite_orm.h>
+#include <catch2/catch_all.hpp>
+#include <cstdio>  //  std::remove
+#if SQLITE_ORM_HAS_INCLUDE(<filesystem>)
+#include <filesystem>
+#endif
+#include "catch_matchers.h"
+
+#ifdef SQLITE_ORM_CTAD_SUPPORTED
+
+struct User {
+    std::string id;
+};
+
+using namespace sqlite_orm;
+
+static const auto default_table = make_table("users", make_column("id", &User::id, primary_key()));
+
+TEST_CASE("vfs modes open successfully") {
+
+#if defined(SQLITE_ORM_APPLE)
+    internal::string_constant_type vfs =
+        GENERATE(unix_vfs_name, unix_posix_vfs_name, unix_dotfile_vfs_name, unix_afp_vfs_name);
+#elif defined(SQLITE_ORM_UNIX)
+    internal::string_constant_type vfs = GENERATE(unix_vfs_name, unix_posix_vfs_name, unix_dotfile_vfs_name);
+#elif defined(SQLITE_ORM_WIN)
+    internal::string_constant_type vfs = GENERATE(win32_vfs_name, win32_longpath_vfs_name);
+#endif
+
+    connection_control options{true, std::string(vfs)};
+    auto storage = make_storage(":memory:", options, default_table);
+    UNSCOPED_INFO("FAILED VFS: " << vfs);
+    REQUIRE_NOTHROW(storage.open_forever());
+
+    REQUIRE(storage.vfs_name() == vfs);
+    REQUIRE(storage.open_mode() == db_open_mode::default_);
+
+    SECTION("Storage copy operator carries over vfs option") {
+        auto storage_copy = storage;
+        CHECK(storage_copy.is_opened());
+        CHECK(storage_copy.vfs_name() == vfs);
+        CHECK(storage_copy.open_mode() == db_open_mode::default_);
```

Comment:
I'd put `REQUIRE` here cause `REQUIRE` fails test and we really expect these values to be true